app/src/main/kotlin/com/wire/android/ui/common/ClickableText.kt

```diff
@@ -0,0 +1,139 @@
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.style.TextOverflow
+
+@Composable
+fun ClickableText(
+    text: AnnotatedString,
+    modifier: Modifier = Modifier,
+    color: Color = Color.Unspecified,
+    textDecoration: TextDecoration? = null,
+    textAlign: TextAlign? = null,
+    overflow: TextOverflow = TextOverflow.Clip,
+    softWrap: Boolean = true,
+    maxLines: Int = Int.MAX_VALUE,
+    onTextLayout: (TextLayoutResult) -> Unit = {},
+    style: TextStyle = LocalTextStyle.current,
+    onClick: (Int) -> Unit,
+    onLongClick: (() -> Unit)? = null,
+) {
+    val layoutResult = remember { mutableStateOf<TextLayoutResult?>(null) }
+
+    // to provide current lambdas and states otherwise it will use the ones it got when starting a LaunchedEffect inside pointerInput
+    val currentOnClick by rememberUpdatedState(newValue = onClick)
+    val currentOnLongClick by rememberUpdatedState(newValue = onLongClick)
+    val currentLayoutResult by rememberUpdatedState(layoutResult)
+
+    val pressIndicator = Modifier.pointerInput(Unit) {
+        detectTapGestures(
+            onTap = { pos ->
+                currentLayoutResult.value?.let { layoutResult ->
+                    currentOnClick(layoutResult.getOffsetForPosition(pos))
+                }
+            },
+            onLongPress = { currentOnLongClick?.invoke() }
+        )
+    }
+    Text(
+        text = text,
+        modifier = modifier.then(pressIndicator),
+        color = color,
+        textDecoration = textDecoration,
+        textAlign = textAlign,
+        overflow = overflow,
+        softWrap = softWrap,
+        maxLines = maxLines,
+        onTextLayout = {
+            layoutResult.value = it
+            onTextLayout(it)
+        },
+        style = style
+    )
+}
+
+data class LinkInfo(
+    val url: String,
+    val start: Int,
+    val end: Int
+)
+
+class SpannableStr(source: CharSequence) : SpannableString(source) {
+    companion object {
+        fun getLinkInfos(text: String, mask: Int): List<LinkInfo> {
+            val spannableStr = SpannableStr(text)
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                Linkify.addLinks(spannableStr, mask) { str: String -> URLSpan(str) }
+            } else {
+                Linkify.addLinks(spannableStr, mask)
+            }
+            return spannableStr.linkInfos
+        }
+    }
+
+    private inner class Data(
+        val what: Any?,
+        val start: Int,
+        val end: Int
+    )
+
+    private val spanList = mutableListOf<Data>()
```

Comment:
We could already collect only links, we could remove `Data` class and re-use `LinkInfo` here, because this `spanList` is only used to get `linkInfos` so we could already have `private val linkInfos = mutableListOf<LinkInfo>()`, that would simplify it, what do you think?